spring-cloud-stream-module-launcher/src/main/java/org/springframework/cloud/stream/module/launcher/ModuleJarLauncher.java

```diff
@@ -47,6 +49,16 @@ public void launch(String[] args) {
 		super.launch(args);
 	}
 
+	/**
+	 * Create and return a new ClassLoader suitable for running the module, or accessing resources
+	 * in it. Disposing of the ClassLoader is the responsibility of the caller.
+	 */
+	public ClassLoader createClassLoader() throws Exception {
```

Comment:
I understand that this is a convenient place for this method because of the underlying infrastructure in place for launching jars. However I believe this method is out of place here. Instances of this class are supposed to be created for launching a jar, not for creating classloaders (hence the name of the class `ModuleJarLauncher`. Perhaps the parts that are required in `ModuleConfigurationMetadataResolver` could be extracted and placed there? I'll make a suggestion for us to pick at above.